clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/DownloadStrategyResolver.ts

```diff
@@ -11,22 +11,30 @@
  * Contributors:
  *   Broadcom, Inc. - initial API and implementation
  */
+import { E4E } from "../../../type/e4eApi";
 import { SettingsService } from "../../Settings";
 import { CopybookName } from "../CopybookDownloadService";
+import { E4ECopybookService } from "../E4ECopybookService";
 import { CopybookDownloaderForDsn } from "./CopybookDownloaderForDsn";
 import { CopybookDownloaderForUss } from "./CopybookDownloaderForUss";
+import { CopybookDownloaderForE4E } from "./CopybookDownloaderForE4E";
 
 /**
- * Copybook downloader from MVS or USS using Zowe Explorer based on configuration
+ * Copybook downloader from E4E Api or using Zowe Explorer MVS & USS Api based on configuration
  */
 export class DownloadStrategyResolver {
   private readonly dsnDownloader?: CopybookDownloaderForDsn;
   private readonly ussDownloader?: CopybookDownloaderForUss;
+  private readonly e4eDownloader?: CopybookDownloaderForE4E;
 
   constructor(
     storagePath: string,
     explorerApi: IApiRegisterClient | undefined,
+    e4eApi: E4E | undefined,
   ) {
+    if (e4eApi) {
+      this.e4eDownloader = new CopybookDownloaderForE4E(storagePath, undefined);
```

Comment:
The `DownloadStrategyResolver` constructor should take optional channel and forward it to the `CopybookDownloaderForE4E`